projects/opentrain/gtfs_pipeline/heatmap/HeatMap.java

```diff
@@ -0,0 +1,287 @@
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  public void draw_map() {
+
+    // color1 to color2
+    Color c1 = new Color(255, 208, 235);
+    Color c2 = new Color(126, 0, 67);
+
+    // Normalize data
+    int maxData = 0;
+    int minData = heatMap[10][10];
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (maxData < heatMap[i][j]) {
+          maxData = heatMap[i][j];
+        }
+        if (heatMap[i][j] != 0 && minData > heatMap[i][j]) {
+          minData = heatMap[i][j];
+        }
+      }
+    }
+
+    // Create image
+    /** TODO: Try different size of image */
+    final BufferedImage image =
+        new BufferedImage(
+            heatMap[0].length - 1000, heatMap.length - 1000, BufferedImage.TYPE_INT_ARGB);
+    final Graphics2D graphics2D = image.createGraphics();
+
+    // Draw the map
+    for (int i = 0; i < heatMap.length - 1; i++) {
+      for (int j = 0; j < heatMap[0].length - 1; j++) {
+        if (heatMap[i][j] == 0) {
+          System.out.println("ERROR:" + "\n" + "heatmap matrix value should not be 0");
+        }
+
+        float value = (float) (heatMap[i][j] - minData) / (float) (maxData - minData);
+
+        int r = (int) (((float) c2.getRed() - (float) c1.getRed()) * value + (float) c1.getRed());
+        int g =
+            (int) (((float) c2.getGreen() - (float) c1.getGreen()) * value + (float) c1.getGreen());
+        int b =
+            (int) (((float) c2.getBlue() - (float) c1.getBlue()) * value + (float) c1.getBlue());
+
+        Color c = new Color(r, g, b);
+        graphics2D.setPaint(c);
+        graphics2D.fillRect(i, j, 2, 2);
+      }
+    }
+
+    graphics2D.dispose();
+    // Save file
+    boolean createImage = false;
+    try {
+      createImage = ImageIO.write(image, "png", new File(imagePath.get()));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    if (!createImage) {
+      System.out.println("can't create image");
+    }
+  }
+
+  public static void main(String[] args) throws IOException {
+
+    Flags.parseCurrentPackage(args);
+
+    StopTime stopTime;
+    Trip trip;
+
+    try (FileInputStream tripInputStream =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/trips.protobin"));
+        FileInputStream stopTimeInputStream =
+            new FileInputStream(new File("projects/opentrain/gtfs_pipeline/stopTime.protobin"))) {
+
+      // Test some trip stops heatMap
+      trip = Trip.parseDelimitedFrom(tripInputStream);
+      stopTime = StopTime.parseDelimitedFrom(stopTimeInputStream);
+      while (true) {
+        if (stopTime == null | trip == null) {
+          break;
+        }
+
+        if (trip.getTripId().equals(stopTime.getTripId())
+            && Integer.parseInt(stopTime.getStopSequence()) == 1) {
+          HeatMap hm = new HeatMap(stopTime.getStopId(), stopTime.getDepartureTime());
```

Comment:
Please use full names for variables: `hm->heatMap`. It's more readable that way.